rotkehlchen/chain/evm/decoding/beefy_finance/utils.py

```diff
@@ -0,0 +1,46 @@
+import logging
+
+from rotkehlchen.errors.misc import RemoteError, UnableToDecryptRemoteData
+from rotkehlchen.errors.serialization import DeserializationError
+from rotkehlchen.globaldb.cache import globaldb_set_general_cache_values
+from rotkehlchen.globaldb.handler import GlobalDBHandler
+from rotkehlchen.logging import RotkehlchenLogsAdapter
+from rotkehlchen.serialization.deserialize import deserialize_evm_address
+from rotkehlchen.types import CacheType, ChainID
+from rotkehlchen.utils.network import request_get
+
+from .constants import CHAIN_ID_TO_BEEFY_CHAIN
+
+logger = logging.getLogger(__name__)
+log = RotkehlchenLogsAdapter(logger)
+
+
+def query_beefy_vaults(chain: ChainID) -> None:
+    """Fetches Beefy Finance vault addresses for the given chain and caches them locally.
+
+    It pulls all vaults from the Beefy API, filters for the specified chain,
+    and stores the vault contract addresses in the global database for later use.
+    """
+    vaults, beefy_chain_name = [], CHAIN_ID_TO_BEEFY_CHAIN[chain]
+    try:
+        result = request_get(url='https://api.beefy.finance/vaults')
+        for entry in result:
+            try:
+                if entry['chain'] != beefy_chain_name:
```

Comment:
I missed this before but you are querying the same data serveral times per chain. I checked and it seems that there isn't a way to filter by chain so the api returns it filtered. Can you ask the beefy team about this and create a note? Seems that it can be improved